### ISSUE TITLE
`fsspec` module dependency is missing

### ISSUE THREAD

**Issue by JPvRiel** [modin-project/modin#3504]
### System information

- Ubuntu 20.04
- 0.11.0
- 3.9.7

### Describe the problem

When using `pd.read_parquet(<filename>)`, the following error is caused by a silent dependency:

```
ModuleNotFoundError: No module named 'fsspec'
```

A simple fix is to manually pip install fsspec, but perhaps fsspec should rather be explicitly included in the dependencies stated in setup.py?

**Comment by devin-petersohn** [comment]
Hi @JPvRiel thanks for the report!

Similar to pandas, we try to reduce the number of dependencies in setup.py. pandas also relies on this but doesn't have it in the setup.py, despite relying on it for some reading. 

We definitely shouldn't have a `ModuleNotFoundError` get thrown, maybe a prompt instead? We could add `fsspec` to setup.py but not everyone needs it, which is why we don't currently have it in the setup.py.

**Comment by JPvRiel** [comment]
@devin-petersohn thanks, sure, makes sense to me now since fsspec might only be needed when a few users load data, and even pandas reports a customized import error if one hasn't installed one of the needed libs. E.g. this is how pandas does it for parquet support:

```
ImportError: Unable to find a usable engine; tried using: 'pyarrow', 'fastparquet'.
A suitable version of pyarrow or fastparquet is required for parquet support.
Trying to import the above resulted in these errors:
 - Missing optional dependency 'pyarrow'. pyarrow is required for parquet support. Use pip or conda to install pyarrow.
 - Missing optional dependency 'fastparquet'. fastparquet is required for parquet support. Use pip or conda to install fastparquet
```

I've go no experience packing python libs, but seems like `extras_require` is the common place to do it with setuptools?

Maybe this? <https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies>

**Comment by devin-petersohn** [comment]
Thanks @JPvRiel that's helpful. We'll take a look.

**Comment by vnlitvinov** [comment]
Should be closed by #3542 now.
